### DISCUSSION TITLE
Unable to copy a Json value to another table

### DISCUSSION THREAD

**Discussion by naeem-qv** [prisma/prisma#23357] [Q&A] [answered]
### Bug description

As you can see in my schema, I have one user with multiple designs and one design with multiple orders. All I want is to do something like this:
```ts
 const order = await prisma.order.create({
            data:{
                dimension : design.dimension,
                list : design.list,
                designId
            }
        })
```
I want to copy some Json fields to order but I'm getting this warning:
```bash

Type 'JsonValue' is not assignable to type 'JsonNull | InputJsonValue'.  //THIS ERROR IS FOR dimensions which is Json

Type 'JsonValue[]' is not assignable to type 'InputJsonValue[]'.
    Type 'JsonValue' is not assignable to type 'InputJsonValue'. //THESE TWO ERROR ARE FOR list which is Json[]
```

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id        String     @id @default(cuid())
  email     String     @unique
  name      String
  password  String
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt
  designs   Design[]
}

model Design {
  designId     String     @id @default(uuid())
  name         String
  list               Json[]
  dimension    Json
  createdAt    DateTime   @default(now())
  updatedAt    DateTime   @updatedAt
  userId       String
  user         User       @relation(fields: [userId], references: [id])
  Orders    Order[]
}

model Order{
  staticId     String   @id @default(uuid())
  name         String
  list              Json[]
  dimension        Json
  createdAt    DateTime @default(now())
  updatedAt    DateTime @updatedAt
  designId     String
  designs      Design   @relation(fields: [designId], references: [designId])
}

```

```ts
// Add your code using Prisma Client
```

### Environment & setup

Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v20.9.0

### Prisma Version

```
prisma                  : 5.10.2
@prisma/client          : 5.10.2
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v20.9.0
```

**Comment by ludralph** [comment, marked as answer]
Hi @naeem-qv 👋 

Thank you for raising this question.

Would casting it to `Prisma.InputJsonValue` resolve the issue for you?

```ts
const order = await prisma.order.create({
  data: {
    dimension: design.dimension as Prisma.InputJsonValue,
    list: design.list as Prisma.InputJsonValue[],
    designId: designId,
  },
});
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by naeem-qv** [reply]
Hi @ludralph 
Thank you for answering my question,
Yes it solved my issue and I can make entries into my db now.
once again thank you

**Reply by golergka** [reply]
I have encountered the same problem and I really don't think this solution is correct.

1. If all possible values of `Prisma.JsonValue` are valid values of `Prisma.InputJsonValue`, then these types should be modified to reflect this, and cast will not be necessary.
2. If some of possible values of `Prisma.JsonValue` are not valid values of `Prisma.InputJsonValue`, then this casting will introduce bugs.

Generally speaking, using casts to solve type problems is a code smell. Can we get a proper way of addressing this?

**Reply by naeem-qv** [reply]
> I have encountered the same problem and I really don't think this solution is correct.
> 
> 1. If all possible values of `Prisma.JsonValue` are valid values of `Prisma.InputJsonValue`, then these types should be modified to reflect this, and cast will not be necessary.
> 2. If some of possible values of `Prisma.JsonValue` are not valid values of `Prisma.InputJsonValue`, then this casting will introduce bugs.
> 
> Generally speaking, using casts to solve type problems is a code smell. Can we get a proper way of addressing this?

Yep casting here feels like a dirty fix to me as well but it's working fine....

**Comment by simoami** [comment]
Instead of casting, you can add a JsonNull as fallback. This will satisfy the missing null option:
```ts
dimension: design.dimension || Prisma.JsonNull
```
